lib/common/options.c

```diff
@@ -939,68 +939,80 @@ pcmk__cluster_option(GHashTable *options, const char *name)
  * \internal
  * \brief Add a description element to a meta-data string
  *
- * \param[in,out] s       Meta-data string to add to
+ * \param[in,out] parent  Parent XML node
  * \param[in]     tag     Name of element to add (\c PCMK_XE_LONGDESC or
  *                        \c PCMK_XE_SHORTDESC)
  * \param[in]     desc    Textual description to add
  * \param[in]     values  If not \p NULL, the allowed values for the parameter
- * \param[in]     spaces  If not \p NULL, spaces to insert at the beginning of
- *                        each line
+ *
+ * \return Standard Pacemaker return code
  */
-static void
-add_desc(GString *s, const char *tag, const char *desc, const char *values,
-         const char *spaces)
+static int
+add_desc(xmlNode *parent, const char *tag, const char *desc, const char *values)
 {
-    char *escaped_en = crm_xml_escape(desc);
-
-    if (spaces != NULL) {
-        g_string_append(s, spaces);
-    }
-    pcmk__g_strcat(s,
-                   "<", tag, " " PCMK_XA_LANG "=\"" PCMK__VALUE_EN "\">",
-                   escaped_en, NULL);
+    const char *full_desc = desc;
+    xmlNode *node = NULL;
+    char *desc_with_values = NULL;
+    int rc = pcmk_rc_ok;
 
     if (values != NULL) {
-        // Append a period if desc doesn't end in "." or ".)"
-        if (!pcmk__str_empty(escaped_en)
-            && (s->str[s->len - 1] != '.')
-            && ((s->str[s->len - 2] != '.') || (s->str[s->len - 1] != ')'))) {
+        // @TODO Use XML node for allowed values
```

Comment:
On second thought, the `<option>` elements that get created for `"select"`-type options make the `"Allowed values: "` string section redundant. We could probably drop it and make this `add_desc()` function much simpler.

There's at least one non-`select` option (`cluster-recheck-interval`) that has an allowed values string. It's not a list of enumerated values. For any options like that, I can just copy the allowed values string into the longdesc. That's more appropriate.

Reply:
Done